### PR TITLE
Fixes #205 - Docker install not finding GPUs

### LINKED ISSUE #205: Docker install not finding GPUs
On a p3.2xlarge, the GPU Docker install (built with the `--gpu` flag) is not finding GPU devices. From `ts_log.log` - note the "Number of GPUs" line:

```
model-server@a8819511cfaa:~/logs$ more ts_log.log
2020-04-16 01:53:17,099 [INFO ] main org.pytorch.serve.ModelServer -
TS Home: /usr/local/lib/python3.6/dist-packages
Current directory: /home/model-server
Temp directory: /home/model-server/tmp
Number of GPUs: 0
Number of CPUs: 8
Max heap size: 13644 M
Python executable: /usr/bin/python3
Config file: /home/model-server/config.properties
Inference address: http://0.0.0.0:8080
Management address: http://0.0.0.0:8081
Model Store: /home/model-server/model-store
Initial Models: N/A
Log dir: /home/model-server/logs
Metrics dir: /home/model-server/logs
Netty threads: 32
Netty client threads: 0
Default workers per model: 8
Blacklist Regex: N/A
Maximum Response Size: 6553500
Maximum Request Size: 6553500
```

When starting on a GPU machine with CUDA drivers installed, I normally see `Number of GPUs: 4`.

### PR DESCRIPTION
**Pull request by dhanainme**
Fixes #205 - Docker install not finding GPUs

* Invokes docker with  --runtime=nvidia for GPU
* Added the option to specify the GPU / specific GPU ids in start.sh
* Fixed the documentation for start.sh script
* Fixed the JDK version in Dockerfile.gpu

Tests : 

Testing : This was tested on a DL AMI 27, Ubuntu 18 / p3.8xlarge (which has 4 CUDA-compatible GPUs). I built the Docker image with the --gpu directive.

Terminal Output for all 3 cases :

```
ubuntu@ip-172-31-32-255:~/serve$ ./start.sh
Starting pytorch/torchserve:latest-gpu docker image
Successfully started torchserve in docker
Registering resnet-18 model
.....
ubuntu@ip-172-31-32-255:~/serve$ docker exec -it 4a162a13f695 head logs/ts_log.log
....
Temp directory: /home/model-server/tmp
Number of GPUs: 0
Number of CPUs: 32

ubuntu@ip-172-31-32-255:~/serve$ ./start.sh -g
Starting pytorch/torchserve:latest-gpu docker image
Successfully started torchserve in docker
Registering resnet-18 model
....
ubuntu@ip-172-31-32-255:~/serve$ docker exec -it 281fc5744248 head logs/ts_log.log
....
Temp directory: /home/model-server/tmp
Number of GPUs: 4
Number of CPUs: 32

ubuntu@ip-172-31-32-255:~/serve$ ./start.sh --gpu --gpu_devices 1,2,3
Starting pytorch/torchserve:latest-gpu docker image
Successfully started torchserve in docker
Registering resnet-18 model
successfully registered resnet-18 model with torchserve
...

ubuntu@ip-172-31-32-255:~/serve$ docker exec -it e05ff19681f9 head logs/ts_log.log
...
Number of GPUs: 3
Number of CPUs: 32
...
```

### CODE REVIEW

File: start.sh [hunk 1]
```diff
@@ -1,9 +1,35 @@
...
+    case $arg in
+        -h|--help)
+          echo "options:"
+          echo "-h, --help  show brief help"
+          echo "-g, --gpu specify to use gpu"
+          echo "-d, --gpu_devices to use specific gpu device ids"
+          exit 0
+          ;;
+        -g|--gpu)
+          DOCKER_RUNTIME="--runtime=nvidia"
```
Comment (chauhang) [inline start.sh:15]: @dhanainme The new way of running docker with gpus is with --gpus flag. Please change to that as described in https://github.com/pytorch/serve/issues/191

File: start.sh [hunk 2]
```diff
@@ -1,9 +1,35 @@
...
+          IMAGE_NAME="pytorch/torchserve:latest-gpu"
+          shift
+          ;;
+	-d|--gpu_devices)
```
[the remark this answers is not in the document]
Comment (fbbradheintz) [reply start.sh:19]: Also: Should there be an example command line that shows what a device specifier looks like for this command line? Is it `0`? `cuda:0`?

[the remark this answers is not in the document]
Comment (dhanainme) [reply start.sh:19]: > Shouldn't -d imply -g?

Yes. But I presumed this is understood implicitly. Have fixed this. 

> Have you tested all combinations of options?

Yes. I had tested CPU only, GPU w/o devices flag. GPU with devices flag (with different set of devices to confirm that it works)

> Also: Should there be an example command line that shows what a device specifier looks like for this command line? Is it 0? cuda:0?

The usage is clearly specified in the README.md

File: start.sh (hunk 1 above)
Comment (dhanainme) [reply start.sh:15]: Was trying to use this originally, But the new format uses a very weird quoting for multiple GPUs which does not play well with bash. Would prefer leaving it this way.

docker run --rm -it --gpus '"device=0,1,2"' -p 8080:8080 -p 8081:8081 torchserve:v0.1-gpu-latest

File: start.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (dhanainme) [reply start.sh:19]: > serve/docker/README.md is missing any mention of the docker command-line flags for specifying GPU runtime and specific devices - this needs to be corrected.

Have added this. 

> serve/docker/README.md has no mention of the build_image.sh and start.sh scripts. Was this intentional? If not, please update the Docker README.

I think its intentional. The start.sh & build_image.sh are more off a quick start guide for folks trying to get things up & running quickly with out even using docker commands. 

Infact build_image.sh is just a wrapper for **docker build** & start.sh is a wrapper for **docker run** both of which are explained with the right options in serve/docker/README.md

**Comment by dhanainme** [comment]
Too many merge conflicts / updates. Deleting this branch for sanity.

Tracked in https://github.com/pytorch/serve/pull/262. All the feedback from this PR have been fixed.

File: start.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (fbbradheintz) [reply start.sh:19]: Also: Should there be an example command line that shows what a device ID for a GPU looks like.

**Pull request closed without merging**